puts/harness/src/lib.c

```diff
@@ -0,0 +1,16 @@
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+void _log(void (*logger)(const char *), const char *format, ...)
+{
+    char *message = NULL;
+    va_list args;
+
+    va_start(args, format);
+    vasprintf(&message, format, args);
```

Comment:
Can this allocation here fail?

Reply:
Yes. In that case the message will be NULL, which must be handled by the logger callback.

I purposely didn't check for NULL here, because I don't believe a correct decision can be made at this level on how to handle this case: it might be because something went wrong in the vendor library that we cannot allocate memory. In that case, we don't want to fail here in the harness but rather at a higher level in Rust or when giving control back to the vendor library, so that we don't mistake a library bug for a harness bug.